edisgo/tools/tools.py

```diff
@@ -406,3 +424,64 @@ def assign_voltage_level_to_component(edisgo_obj, df):
         axis=1,
     )
     return df
+
+
+def get_weather_cells_for_grid_district(edisgo_obj):
+    """
+    Get all generator weather cells and weather cells that intersect with 
+    the grid district 
+    
+    Parameters
+    ----------
+    edisgo_obj : :class:`~.EDisGo`
+
+    Returns
+    -------
+    set
+        Set with weather cell IDs
+
+    """
+    Session = sessionmaker(bind=connection(readonly=True))
+    session = Session()
+
+    # Download geometries of weather cells to make sure all weather
+    # cells are loaded accordingly
+    srid = edisgo_obj.topology.grid_district["srid"]
+    table = climate.Cosmoclmgrid
+    query = session.query(
+        table.gid,
+        func.ST_AsText(
+            func.ST_Transform(
+                table.geom, srid
+            )
+        ).label("geometry")
+    )
+
+    geom_data = pd.read_sql_query(
+        query.statement, query.session.bind)
+    geom_data.geometry = geom_data.apply(
+        lambda _: wkt_loads(_.geometry), axis=1)
+    geom_data = gpd.GeoDataFrame(
+        geom_data, crs=f"EPSG:{srid}")
+
+    mv_geom = edisgo_obj.topology.grid_district["geom"]
+
+    # Make sure MV Geometry is MultiPolygon
+    if mv_geom.geom_type == "Polygon":
+        # Transform Polygon to MultiPolygon and overwrite geometry
+        p = wkt_loads(str(mv_geom))
+        m = MultiPolygon([p])
+        edisgo_obj.topology.grid_district["geom"] = m
+    elif mv_geom.geom_type == "MultiPolygon":
+        m = mv_geom
+    else:
+        raise ValueError(
+            f"Grid district geometry is of type {type(mv_geom)}."
+            " Only Shapely Polygon or MultiPolygon are accepted.")
+
+    mv_geom_gdf = gpd.GeoDataFrame(
+        m, crs=f"EPSG:{srid}", columns=["geometry"])
+
+    return set(np.append(gpd.sjoin(
+        geom_data, mv_geom_gdf, how="right", op='intersects').gid.unique(),
+        edisgo_obj.topology.generators_df.weather_cell_id.dropna().unique()))
```

Comment:
Why do you append existing weather cell IDs?

Reply:
Sometimes existing generators had different weather cell IDs than I got from the intersection. I don't know why. But without this append I missed some weather cells and got errors when importing scenarios in some grids.